collections/_development_standards/patterns.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: default
+title:  "Patterns"
+category: development_standards
+---
+
+## Development Patterns
+
+Below are described the major engineering patterns currently in-use across the Service. There is no definitive list as it is understood that patterns should be used when and where appropriate.
+
+### Orchestrator
+
+> At times an orchestrator may be unnecessary for a light weight controller action and it may be easier and less complex to omit the orchestrator completely and instead inject Mediatr directly into the controller.
+
+As controllers age they are asked to do more and more work which can result in serious bloat. We want to keep our controllers focused on what they should be doing such as handling requests, invoking business rules and returning responses. So to keep them light we need to hand-off any other work that may end up in a controller out to somewhere else and the Orchestrator can be used to fill this need. 
+
+The use of Orchestrators can also make a codebase more testable by moving logic and flow out of controllers and their tight coupling to the runtime.
+
+![Orchestrator Overview](./images/OrchestratorOverview.png)
+
+#### Example
+
+    public class ProductsController : Controller
+    {
+        private readonly IProductOrchestrator _orchestrator;
+
+        public ProductsController(IProductOrchestrator orchestrator)
+        {
+            _orchestrator = orchestrator;
+        }
+        
+        public ActionResult Index()
+        {
+            var productsResult = _orchestrator.GetProducts();
+
+            return View(new ProductsModel
+            {
+                Products = productsResult.Products
+            });
+        }
+    }
+
+    public class ProductOrchestrator : IProductOrchestrator
+    {
+        private readonly IMediator _mediator;
+
+        public ProductOrchestrator(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+        
+        public IList<Product> GetProducts()
+        {
+            return = _mediator.Send(new GetProductsQuery());            
+        }
+    }
+
+
+#### References
+
+* [Simplify Your Architecture Using the Orchestrator Pattern](http://www.jamiemaguire.net/index.php/2017/05/06/simplify-your-architecture-using-the-orchestrator-pattern/)
+* [The Orchestrator Pattern](http://www.michaeltaylorp3.net/orchestrator-pattern/)
+* [Never Mind the Controller, Here is the Orchestrator](https://www.simple-talk.com/dotnet/asp-net/never-mind-the-controller-here-is-the-orchestrator/)
+  
+### Mediator
+
+With the mediator pattern, communication between objects is encapsulated within a mediator object and do not communicate directly with each other, but instead communicate through the mediator. This reduces the dependencies between communicating objects and reduces coupling.
```

Comment:
Can we add a comment that the Mediator pattern should only be used to decouple the different layers of the application as opposed to using it to reuse code within the same layer?

Reply:
How does this sound:

`When seeking to use the Mediatr pattern you should look to use it as a mechanism to decouple the different application layers and resist the urge to use it as a code-reuse pattern for which other approaches are more appropriate.`